modules/recovery_vault/recovery_vault.tf

```
@@ -21,7 +21,24 @@ resource "azurerm_recovery_services_vault" "asr" {
   storage_mode_type   = try(var.settings.storage_mode_type, "GeoRedundant")
 
   identity {
-    type = "SystemAssigned"
+    type = try(var.settings.identity.type, "SystemAssigned")
+
+    # if type contains UserAssigned, `identity_ids` is mandatory
+    identity_ids = try(regex("UserAssigned", var.settings.identity.type), null) != null ? flatten([
+      for managed_identity in var.settings.identity.managed_identities : [
+        var.managed_identities[try(managed_identity.lz_key, var.client_config.landingzone_key)][managed_identity.key].id
+      ]
+    ]) : null
   }
 
+  dynamic "encryption" {
+    for_each = can(var.settings.encryption) ? [1] : []
+
+    content {
+      key_id                            = var.settings.encryption.key_id
```

Lets find a way to include the key object model as well as the ```key_id``` litterals.
```suggestion
      key_id                            = can(var.settings.encryption.key_id) ? var.settings.encryption.key_id : var.keyvaultkeys...
```

---

I tried and unfortunately, terraform gives me a circular reference error